crates/node/service/src/actors/mod.rs

```diff
@@ -0,0 +1,31 @@
+//! [NodeActor] services for the node.
+//!
+//! [NodeActor]: super::NodeActor
+
+use async_trait::async_trait;
+
+mod derivation;
+pub use derivation::{DerivationActor, DerivationError, InboundDerivationMessage};
+
+mod l1_watcher_rpc;
+pub use l1_watcher_rpc::{L1WatcherRpc, L1WatcherRpcError};
+
+/// The [NodeActor] is an actor-like service for the node.
+///
+/// Actors may:
+/// - Handle incoming messages.
+///     - Perform background tasks.
+/// - Emit new events for other actors to process.
+#[async_trait]
+pub trait NodeActor {
+    /// The event type received by the actor.
+    type InboundEvent;
+    /// The error type for the actor.
+    type Error;
+
+    /// Starts the actor.
+    async fn start(self) -> Result<(), Self::Error>;
+
+    /// Processes an incoming message.
+    async fn process(&mut self, msg: Self::InboundEvent) -> Result<(), Self::Error>;
+}
```

Comment:
Think I prefer trait definitions in their own trait module to make it clear to the file system enjoyer where things live

Reply:
In the ideal case, mods contain nothing but re-exports

Reply:
Yeah sounds good - figured it was small enough to not warrant a new file, but not super opinionated